wiki/development/arma-3-scheduler-and-our-practices.md

```diff
@@ -10,87 +10,99 @@ order: 8
+### 1.3 Unscheduled Environment
+Code running in the unscheduled environment uses linear execution, that means it will always run from start to finish within one frame, without any pauses. The following will always be executed in the unscheduled environment:
 
-* perFrameHandler
-* Extended_EventHandlers
-* addEventHandler
+* Event Handlers
+* CBA Per Frame Handlers (PFHs)
+* CBA Extended Event Handlers (XEHs)
+* `call` from a function running in *unscheduled* environment
 
 
 ## 2. What is the scheduler and why do I care?
-
 BIKI Article: [https://community.bistudio.com/wiki/Biki2.0:Performance_Considerations](https://community.bistudio.com/wiki/Biki2.0:Performance_Considerations)
 
-The Arma 3 script scheduler basically gives a fair-share execution to all running scripts, FSMs, and SQS files running on any given client or server at any given time. See the BIKI article for a in-depth explanation of this. What this basically means though, is that all scripts get a fair share; this also means scheduled execution is drastically affected by other poorly written mods. For example, if 2 different spawn's are running in a tight loop of `do { foo } while (true);`, they will both get exactly 50% of the scheduling time. 
-
-With the way mission makers and mod makers generally use spawn/execVM, this means you're actually getting drastically less execution time in the scheduled environment than you might think. This leads to visible delay issues all the way up to massive delay on execution. You can easily test and prove this by looping spawns and watching the execution times extend. 
-
-What does this all mean? It means we need to live outside of the spawn execution as much as possible. There are 2 places that the majority of our functionality should stem from, which means that as long as we strictly always perform calls between functions, we are executing within the same frame. If our execution is either stemming from the perFrameHandlers, or any default Extended_EventHandlers, than we can guarantee execution within a single frame. *ANY OTHER CIRCUMSTANCE IS NOT GUARANTEED.*
-
-The scheduler will also actually halt your script mid-execution, usually at the end of a given control block, and pause you to yield to other scripts. This can lead to drastically incorrect results when performing calculations. Again, this is the reason we want all our given code to run to completion in a single given frame.
+The Arma 3 script scheduler basically gives a fair-share execution to all running scripts, FSMs, and SQS files running on any given client or server at any given time. See the BIKI article for a in-depth explanation of this. What this basically means though, is that all scripts get a fair share; this also means scheduled execution is drastically affected by other mods that use scheduled execution. For example, if 2 different spawn's are running in a tight loop of `while {true} do {...};`, they will both get exactly 50% of the scheduling time.
```

Comment:
Could you link the BIKI article mentioned here using this pattern:
`[text](url){:target="_blank"}`